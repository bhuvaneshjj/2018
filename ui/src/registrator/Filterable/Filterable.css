.Filterable_input {
  padding: 1ex;
  margin-bottom: 1ex;
}

.Filterable_zobrazeno {
  margin-left: 2em;
}

.Filterable_kategorie_vykonu {
  margin-left: 2em;
}