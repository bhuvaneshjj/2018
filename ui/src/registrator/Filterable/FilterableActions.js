export const kategorieFilterChange = (actionPrefix, typKategorie) => ({
  type: `${actionPrefix}_KATEGORIE_FILTER_CHANGE`,
  typKategorie
});

export const textFilterChange = (actionPrefix, textFilter) => ({
  type: `${actionPrefix}_TEXT_FILTER_CHANGE`,
  textFilter
});
