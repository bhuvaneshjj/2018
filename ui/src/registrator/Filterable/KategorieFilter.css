.KategorieFilter {
  margin-right: 5px;
}

.KategorieFilter > .active {
  font-weight: bold;
  border-radius: 5px;
  border: 3px solid black !important;
}
