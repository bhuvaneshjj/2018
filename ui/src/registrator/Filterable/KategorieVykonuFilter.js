import React from 'react';
import PropTypes from 'prop-types';
import { Button, Glyphicon } from 'react-bootstrap';
import { barvaProTypKategorie } from '../../Util';
import './KategorieVykonuFilter.css';

const KategorieVykonuFilter = ({ typKategorie, active, onClick }) => {
  const style = {
    backgroundColor: barvaProTypKategorie(typKategorie),
    fontWeight: active ? 'bold' : '' // override style from <Button/>
  };
  return (
    <span className={`KategorieVykonuFilter_${active ? 'active' : 'nonactive'}`}>
      <Button onClick={onClick} style={style}>
        <Glyphicon glyph="filter" /> {typKategorie}
      </Button>
    </span>
  );
};

KategorieVykonuFilter.propTypes = {
  typKategorie: PropTypes.string.isRequired,
  active: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default KategorieVykonuFilter;
