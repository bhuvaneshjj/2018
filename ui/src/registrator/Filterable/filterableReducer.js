export const initialState = {
  kategorieFilter: '',
  textFilter: ''
};

export const createFilterableReducer = actionPrefix => (state = initialState, action) => {
  switch (action.type) {
    case `${actionPrefix}_KATEGORIE_FILTER_CHANGE`:
      if (state.kategorieFilter === action.typKategorie) {
        return { ...state, kategorieFilter: '' };
      }
      return { ...state, kategorieFilter: action.typKategorie };
    case `${actionPrefix}_TEXT_FILTER_CHANGE`:
      return { ...state, textFilter: action.textFilter.toLowerCase() };
    default:
      return state;
  }
};
