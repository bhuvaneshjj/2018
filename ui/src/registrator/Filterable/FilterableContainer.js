import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createKategorieFilterChange, createTextFilterChange } from './FilterableActions';
import Filterable from './Filterable';

const mapStateToProps = (state, ownProps) => {
  const { reduxName, numberOfItems } = ownProps;
  const { kategorieFilter, textFilter } = state.registrator[reduxName];

  return { kategorieFilter, numberOfItems, textFilter };
};

const mapDispatchToProps = (dispatch, ownProps) => {
  const { actionPrefix } = ownProps;

  return {
    onTextFilterChange: text => dispatch(createTextFilterChange(actionPrefix)(text)),
    onKategorieFilterChange: typKategorie =>
      dispatch(createKategorieFilterChange(actionPrefix)(typKategorie))
  };
};

const FilterableContainer = connect(mapStateToProps, mapDispatchToProps)(Filterable);

FilterableContainer.propTypes = {
  actionPrefix: PropTypes.string.isRequired,
  reduxName: PropTypes.string.isRequired,
  numberOfItems: PropTypes.number.isRequired
};

export default FilterableContainer;
