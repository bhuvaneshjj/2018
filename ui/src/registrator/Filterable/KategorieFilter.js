import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import { barvaProTypKategorie } from '../../Util';
import PopisekKategorie from '../../shared/PopisekKategorie/PopisekKategorie';
import './KategorieFilter.css';

const KategorieFilter = ({ typKategorie, active, onClick }) => {
  const style = {
    backgroundColor: barvaProTypKategorie(typKategorie, active ? '1.0' : '0.5')
  };

  return (
    <span className="KategorieFilter">
      <Button onClick={onClick} style={style} active={active}>
        <PopisekKategorie typ={typKategorie} />
      </Button>
    </span>
  );
};

KategorieFilter.propTypes = {
  typKategorie: PropTypes.string.isRequired,
  active: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default KategorieFilter;
