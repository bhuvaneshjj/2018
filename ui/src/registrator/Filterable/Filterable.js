import React from 'react';
import PropTypes from 'prop-types';
import { Badge, ButtonGroup } from 'react-bootstrap';
import DebounceInput from 'react-debounce-input';
import { TYPY_KATEGORII } from '../../constants';
import KategorieVykonuFilter from './KategorieVykonuFilter';
import './Filterable.css';

const Filterable = ({
  kategorieVykonuFilter,
  numberOfItems,
  textFilter,
  onKategorieVykonuFilterChange,
  onTextFilterChange
}) => (
  <div>
    <DebounceInput
      className="Filterable_input"
      minLength={0}
      debounceTimeout={500}
      value={textFilter}
      placeholder="Filtr na příjmení a jméno"
      onChange={e => onTextFilterChange(e.target.value)}
    />

    <ButtonGroup className="Filterable_kategorie_vykonu">
      {TYPY_KATEGORII.map(typKategorie => (
        <KategorieVykonuFilter
          key={typKategorie}
          typKategorie={typKategorie}
          onClick={() => onKategorieVykonuFilterChange(typKategorie)}
          active={kategorieVykonuFilter === typKategorie}
        />
      ))}
    </ButtonGroup>

    <span className="Filterable_zobrazeno">
      zobrazeno: <Badge>{numberOfItems}</Badge>
    </span>
  </div>
);

Filterable.propTypes = {
  kategorieVykonuFilter: PropTypes.string,
  numberOfItems: PropTypes.number.isRequired,
  textFilter: PropTypes.string,
  onKategorieVykonuFilterChange: PropTypes.func.isRequired,
  onTextFilterChange: PropTypes.func.isRequired
};

export default Filterable;
