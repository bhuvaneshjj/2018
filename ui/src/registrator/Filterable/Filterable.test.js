import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Filterable from './Filterable';

jest.useFakeTimers();

it('renders', () => {
  const wrapper = shallow(
    <Filterable
      numberOfItems={0}
      onKategorieVykonuFilterChange={jest.fn()}
      onTextFilterChange={jest.fn()}
    />
  );
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('maps DebounceInput.onChange to dispatch onTextFilterChange action', () => {
  const onTextFilterChange = jest.fn();
  const wrapper = shallow(
    <Filterable
      numberOfItems={1}
      onKategorieVykonuFilterChange={jest.fn()}
      onTextFilterChange={onTextFilterChange}
    />
  );
  expect(wrapper.find('.Filterable_input')).toHaveLength(1);
  wrapper.find('.Filterable_input').simulate('change', { target: { value: 'K' } });

  jest.runAllTimers();
  expect(onTextFilterChange).toHaveBeenCalledWith('K');
});

it('maps KategorieVykonuFilter.onClick to dispatch onKategorieVykonuFilterChange action', () => {
  const onKategorieVykonuFilterChange = jest.fn();
  const wrapper = mount(
    <Filterable
      numberOfItems={2}
      onKategorieVykonuFilterChange={onKategorieVykonuFilterChange}
      onTextFilterChange={jest.fn()}
    />
  );
  expect(wrapper.find('button')).toHaveLength(5);
  wrapper
    .find('button')
    .first()
    .simulate('click');

  expect(onKategorieVykonuFilterChange).toHaveBeenCalledWith('maraton');
});
