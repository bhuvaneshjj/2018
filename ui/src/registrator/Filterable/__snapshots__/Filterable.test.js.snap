// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <input
    className="Filterable_input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Filtr na příjmení a jméno"
    type="text"
    value=""
  />
  <div
    className="Filterable_kategorie btn-group"
  >
    <span
      className="KategorieFilter"
    >
      <button
        className="KategorieFilter--maraton--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          <img
            alt="maraton"
            height={30}
            src="maraton.png"
            title="maraton"
          />
           
          maraton
        </span>
      </button>
    </span>
    <span
      className="KategorieFilter"
    >
      <button
        className="KategorieFilter--půlmaraton--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          <img
            alt="půlmaraton"
            height={30}
            src="pulmaraton.png"
            title="půlmaraton"
          />
           
          půlmaraton
        </span>
      </button>
    </span>
    <span
      className="KategorieFilter"
    >
      <button
        className="KategorieFilter--cyklo--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          <img
            alt="cyklo"
            height={30}
            src="cyklo.png"
            title="cyklo"
          />
           
          cyklo
        </span>
      </button>
    </span>
    <span
      className="KategorieFilter"
    >
      <button
        className="KategorieFilter--koloběžka--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          <img
            alt="koloběžka"
            height={30}
            src="kolobezka.png"
            title="koloběžka"
          />
           
          koloběžka
        </span>
      </button>
    </span>
    <span
      className="KategorieFilter"
    >
      <button
        className="KategorieFilter--pěší--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          <img
            alt="pěší"
            height={30}
            src="pesi.png"
            title="pěší"
          />
           
          pěší
        </span>
      </button>
    </span>
  </div>
  <span
    className="Filterable_zobrazeno"
  >
    zobrazeno: 
    <span
      className="badge"
    >
      0
    </span>
  </span>
</div>
`;
